util/expiry_os.h

```diff
@@ -40,7 +40,6 @@ class ExpiryModuleOS : public ExpiryModule
     ExpiryModuleOS()
```

Comment:
Comments in your wiki indicate that you already know this, but just in case: you can make this ctor (and that of ExpiryModuleEE) protected and declare ExpiryModule::CreateExpiryModule() as a friend, which forces everyone to allocate via your factory method. The only complication is in the unit tests, where you would need to update them to allocate via the factory method.